runtime/src/bank.rs

```diff
@@ -1211,8 +1204,37 @@ impl Bank {
     }
 
     pub fn add_native_program(&self, name: &str, program_id: &Pubkey) {
-        let account = native_loader::create_loadable_account(name);
-        self.store_account(program_id, &account);
+        let mut already_genuine_program_exists = false;
+        if let Some(mut account) = self.get_account(&program_id) {
+            already_genuine_program_exists = native_loader::check_id(&account.owner);
```

Comment:
What does genuine mean here, could just be `is_native`

Reply:
Good call. :) Not strongly-opinionated word choice, I wanted to emphasize that an native-loader's accounts can be good or bad. And this is really trying to indicate this variable holds a flag for the good native loader program's account.

I thought of these contrast:

real, legitimate, genuine <=> spoofed/crafted/malicious